### PR TITLE
docs(splunk_hec_metrics sink): Use generated docs for configuration

### LINKED ISSUE #16332: Convert the `splunk_hec_metrics` sink to use auto-generated docs (@StephenWakely)

### PR DESCRIPTION
**Pull request by StephenWakely**
Closes #16332 
Epic #14999 

Signed-off-by: Stephen Wakely <fungus.humungus@gmail.com>

### CODE REVIEW

File: website/cue/reference/components/sinks/splunk_hec_logs.cue [hunk 1]
```diff
@@ -72,7 +72,7 @@ components: sinks: splunk_hec_logs: {
 		notices: []
 	}
 
-	configuration: sinks._splunk_hec.configuration & {
```
Comment (StephenWakely) [inline website/cue/reference/components/sinks/splunk_hec_logs.cue:75]: This is only needed until #16334 merges.

File: src/sinks/splunk_hec/common/acknowledgements.rs [hunk 2]
```diff
@@ -32,6 +32,7 @@ pub struct HecClientAcknowledgementsConfig {
     pub indexer_acknowledgements_enabled: bool,
 
     /// The amount of time, in seconds, to wait in between queries to the Splunk HEC indexer acknowledgement endpoint.
```
Comment (spencergilbert) [inline src/sinks/splunk_hec/common/acknowledgements.rs:34]:
```suggestion
    /// The amount of time to wait in between queries to the Splunk HEC indexer acknowledgement endpoint.
```

**Review by spencergilbert — CHANGES_REQUESTED** [review]

**Review by spencergilbert — APPROVED** [review]

File: src/sinks/splunk_hec/metrics/config.rs [hunk 3]
```diff
@@ -57,18 +72,29 @@ pub struct HecMetricsSinkConfig {
     /// The name of the index where to send the events to.
     ///
     /// If not specified, the default index is used.
+    #[configurable(metadata(docs::examples = "{{ host }}", docs::examples = "custom_index"))]
     pub index: Option<Template>,
```
Comment (neuronull) [inline src/sinks/splunk_hec/metrics/config.rs:76]: same question as in the other PR about default

File: src/sinks/splunk_hec/metrics/config.rs [hunk 4]
```diff
@@ -57,18 +72,29 @@ pub struct HecMetricsSinkConfig {
...
 
     /// The sourcetype of events sent to this sink.
     ///
     /// If unset, Splunk will default to `httpevent`.
```
Comment (neuronull) [inline src/sinks/splunk_hec/metrics/config.rs:80]: same ? as other PR, should we just default to httpevent? 🤔

File: src/sinks/splunk_hec/metrics/config.rs [hunk 5]
```diff
@@ -57,18 +72,29 @@ pub struct HecMetricsSinkConfig {
...
+        docs::examples = "_json",
+        docs::examples = "httpevent"
+    ))]
     pub sourcetype: Option<Template>,
 
     /// The source of events sent to this sink.
     ///
     /// This is typically the filename the logs originated from.
     ///
     /// If unset, the Splunk collector will set it.
```
Comment (neuronull) [inline src/sinks/splunk_hec/metrics/config.rs:92]: worthy of an external link? 🤷

File: src/sinks/splunk_hec/metrics/config.rs (hunk 3 above)
Comment (StephenWakely) [reply src/sinks/splunk_hec/metrics/config.rs]: Same as [other PR](https://github.com/vectordotdev/vector/pull/16334#discussion_r1101251795).

File: src/sinks/splunk_hec/metrics/config.rs (hunk 5 above)
Comment (StephenWakely) [reply src/sinks/splunk_hec/metrics/config.rs:88]: Same as [other PR](https://github.com/vectordotdev/vector/pull/16334#discussion_r1101266431).

File: src/sinks/splunk_hec/common/acknowledgements.rs [hunk 6]
```diff
@@ -31,7 +31,8 @@ pub struct HecClientAcknowledgementsConfig {
     /// [splunk_indexer_ack_docs]: https://docs.splunk.com/Documentation/Splunk/8.2.3/Data/AboutHECIDXAck
     pub indexer_acknowledgements_enabled: bool,
 
-    /// The amount of time, in seconds, to wait in between queries to the Splunk HEC indexer acknowledgement endpoint.
+    /// The amount of time to wait in between queries to the Splunk HEC indexer acknowledgement endpoint.
```
Comment (spencergilbert) [inline src/sinks/splunk_hec/common/acknowledgements.rs:34]:
```suggestion
    /// The amount of time to wait between queries to the Splunk HEC indexer acknowledgement endpoint.
```

**Review by spencergilbert — APPROVED** [review]
According to [grammarly](https://www.grammarly.com/blog/in-between-or-inbetween/), but approving the entirety of the PR.

**Pull request merged**